server/market/market_test.go

```diff
@@ -0,0 +1,366 @@
+func (ta *TArchivist) CancelOrder(*order.LimitOrder) error                    { return nil }
+func (ta *TArchivist) RevokeOrder(*order.LimitOrder) error                    { return nil }
+func (ta *TArchivist) FailCancelOrder(*order.CancelOrder) error               { return nil }
+func (ta *TArchivist) UpdateOrderFilled(order.Order) error                    { return nil }
+func (ta *TArchivist) UpdateOrderStatus(order.Order, types.OrderStatus) error { return nil }
+func (ta *TArchivist) UpdateMatch(match *order.Match) error                   { return nil }
+func (ta *TArchivist) MatchByID(mid order.MatchID, base, quote uint32) (*db.MatchData, error) {
+	return nil, nil
+}
+func (ta *TArchivist) UserMatches(aid account.AccountID, base, quote uint32) ([]*db.MatchData, error) {
+	return nil, nil
+}
+func (ta *TArchivist) CloseAccount(account.AccountID, account.Rule)                 {}
+func (ta *TArchivist) Account(account.AccountID) (acct *account.Account, paid bool) { return nil, false }
+func (ta *TArchivist) CreateAccount(*account.Account) (string, error)               { return "", nil }
+func (ta *TArchivist) AccountRegAddr(account.AccountID) (string, error)             { return "", nil }
+func (ta *TArchivist) PayAccount(account.AccountID, string, uint32) error           { return nil }
+
+func randomAccountID() account.AccountID {
+	pk := randomBytes(pki.PubKeySize) // size is not important since it is going to be hashed
+	return account.NewID(pk)
+}
+
+func randomOrderID() order.OrderID {
+	pk := randomBytes(order.OrderIDSize)
+	var id order.OrderID
+	copy(id[:], pk)
+	return id
+}
+
+func TestMarket_runEpochs(t *testing.T) {
+	// This test exercises the Market's main loop, which cycles the epochs and
+	// queues (or not) incoming orders.
+	qty := uint64(dcrLotSize) * 10
+	rate := uint64(1000) * dcrRateStep
+	aid := randomAccountID()
+	limit := &msgjson.Limit{
+		Prefix: msgjson.Prefix{
+			AccountID:  aid[:],
+			Base:       dcrID,
+			Quote:      btcID,
+			OrderType:  msgjson.LimitOrderNum,
+			ClientTime: uint64(time.Now().Unix()),
+		},
+		Trade: msgjson.Trade{
+			Side:     msgjson.SellOrderNum,
+			Quantity: qty,
+			UTXOs:    []*msgjson.UTXO{},
+			Address:  btcAddr,
+		},
+		Rate: rate,
+		TiF:  msgjson.StandingOrderNum,
+	}
+
+	lo := &order.LimitOrder{
+		MarketOrder: order.MarketOrder{
+			Prefix: order.Prefix{
+				AccountID:  aid,
+				BaseAsset:  limit.Base,
+				QuoteAsset: limit.Quote,
+				OrderType:  order.LimitOrderType,
+				ClientTime: time.Unix(int64(limit.ClientTime), 0).UTC(),
+			},
+			UTXOs:    []order.Outpoint{},
+			Sell:     true,
+			Quantity: limit.Quantity,
+			Address:  limit.Address,
+		},
+		Rate:  limit.Rate,
+		Force: order.StandingTiF,
+	}
+
+	oRecord := orderRecord{
+		msgID: 1,
+		req:   limit,
+		order: lo,
+	}
+
+	epochDurationSec := int64(1)
+
+	mkt, err := NewMarket(context.Background(), epochDurationSec, assetDCR, assetBTC, &TArchivist{}, &TAuth{})
```

Comment:
Note that these test are using an actual `Book`, but this was way easier than stubbing a book, and it allows testing the constructor.  `TestMarket_processEpoch` below is a more limited test, but it still uses the constructor.